filament/src/details/Material.cpp

```diff
@@ -91,6 +95,17 @@ Material::Builder& Material::Builder::package(const void* payload, size_t size)
     return *this;
 }
 
+template<typename T, typename>
+Material::Builder& Material::Builder::constant(const char* name, T value) {
+    ASSERT_PRECONDITION(name != nullptr, "name cannot be null");
+    mImpl->mConstantSpecializations[name] = value;
+    return *this;
+}
+
+template Material::Builder& Material::Builder::constant<int32_t>(const char* name, int32_t value);
+template Material::Builder& Material::Builder::constant<float>(const char* name, float value);
+template Material::Builder& Material::Builder::constant<bool>(const char* name, bool value);
```

Comment:
look at `MaterialInstance::setParameter`, these method should take a `const char* name, size_t len` to be compatible with `std::string_view`. It might be good to have a version that takes a `StringLiteral` also.